src/main/scala/where/Reference.scala

```diff
@@ -11,7 +13,7 @@ import com.originate.scalypher.CypherExpressible
 import scala.language.implicitConversions
 
 sealed trait Reference extends ToQueryWithIdentifiers {
-  def getReferenceable: Option[Referenceable] = None
```

Comment:
yeah i didn't like having it default to None, it removes the assurance you get that the compiler will remind me later that "this thing needs to get implemented!" if i add another type of Reference